mitodl/edx-sga
# Compare Only Final Submissions When Report Downloading

#### What are the relevant tickets?
[TNL-7511](https://openedx.atlassian.net/browse/TNL-7511)


#### What's this PR do?
There is a bug in getting students' submissions for comparison with the report generated for submissions.  There are two types of submissions, Finalised (File Uploaded and Submitted) and non-finalized (File Uploaded Only). When you click **Download All Submissions**,
<img width="242" alt="Screen Shot 2021-02-09 at 7 03 26 PM" src="https://user-images.githubusercontent.com/4252738/107374523-7fec4380-6b09-11eb-94ac-c1f1abaaf68d.png">

The system fetches _current_ student submissions and compares them when the submissions created in the last report. If the count differs, the system updates/replaces the generated report. There was a bug in this comparison. The report is generated for both types of submissions while the current/recent submission count is fetched only for **Finalized** submissions due to which the count is never equal and the system updates/replace the file each time the download is requested. 
This PR fixes that anomaly and generates reports only for **Finalized** submissions. 


#### How should this be manually tested?
1. Go to the SGA component, and upload **.pdf** file from up to 3 different accounts. 
2. Submit the problem from 2 of 3 accounts and leave only uploaded. 
3. Click the Grade Submissions tab on the SGA xblock
2. All of the written submissions appear in a table. Verify two submissions appear "Submitted: Yes" and for one "Submitted: No"
3. To download all submissions, click the **Download All Submissions** button at the top.
4. It will give a message in red saying: it is preparing the file of all student submissions.
5. When the file is ready, it will give a message in green saying: Student submission file ready for download
6. Click the Download all submission button again.
Expected behavior: Zip file will download
Actual behavior:  System says it is preparing the file of all student submissions.



#### Screenshots (if appropriate)
<img width="449" alt="Screen Shot 2021-02-09 at 7 12 58 PM" src="https://user-images.githubusercontent.com/4252738/107375690-d4dc8980-6b0a-11eb-9004-ddc4694a2a85.png">

Code Review:
@pdpinch
